src/controller/SetUpCodeParser.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      Implementation of SetUp Code Parser, a class
+ *      that implements parsing manual and QRCode for connecting
+ *      to a CHIP Device.
+ *
+ */
+
+#include <controller/SetUpCodeParser.h>
+
+#include <controller/CHIPDeviceController.h>
+#include <lib/mdns/Resolver.h>
+#include <lib/support/CodeUtils.h>
+
+namespace chip {
+namespace Controller {
+
+CHIP_ERROR SetUpCodeParser::PairDevice(NodeId remoteId, const char * setUpCode)
+{
+    SetupPayload payload;
+
+    bool isQRCode = strncmp(setUpCode, kQRCodePrefix, strlen(kQRCodePrefix)) == 0;
+    ReturnErrorOnFailure(isQRCode ? QRCodeSetupPayloadParser(setUpCode).populatePayload(payload)
+                                  : ManualSetupPayloadParser(setUpCode).populatePayload(payload));
+
+    mRemoteId     = remoteId;
```

Comment:
At the start of commissioning, the commissionee will not yet have a node ID.  What then is this?

Reply:
Then I will just simply call this out as one more path that will eventually require refactor to remove this oddity from the sdk's controller/commissioner, which is at adds with the spec, as the spec establishes no assumptions that a commissioner node will be assigning node IDs.  In fact, this is unlikely in real implementations.